### PR TITLE
Propor type-hinting

### PR DESCRIPTION
**Pull request by schlopp**
Hello! 👋,
I'm currently in the process of resolving type-hinting issues within Acord. This pull request is still in the works, and edits are allowed by maintainers.

### CODE REVIEW

**Comment by Seniatical** [comment]
Looking good so far 👍

**Comment by schlopp** [comment]
My last commit has altered a few things in embeds.py:
- Set defaults to the Embed class
- Clean code
- Alter code to comply with said defaults

Results:
- 3 future bugs resolved
- Better type hinting for anyone using an IDE
- Less memory usage

**Comment by Seniatical** [comment]
with docstrings be careful with "`", wrapping something in singular ones will make the text italic. For like a `box` use "``" instead

File: acord/bases/flags/base.py [hunk 1]
```diff
@@ -28,14 +28,15 @@ def __call__(cls, value: int = 0, **kwds: Dict[str, bool]):
         ----------
         value: :class:`int`
             Value of flag
-        **kwds: Dict[:class:`str`, :class:`bool`]
+        **kwargs: Dict[:class:`str`, :class:`bool`]
             Name of flags you want to use,
             value must be set to ``True``!
         """
+
         enum = super(cls.__class__, cls).__call__(int(value))
```
Comment (Seniatical) [inline acord/bases/flags/base.py:38]: if implementing `*args` possibly add to call to super.

**Pull request closed without merging**